File: DynamicProgramming1/01Tiles/C01Tiles/01Tiles.c
#include <stdio.h>

#define MAX 1000000
#define MOD 15746

int n;
int d[MAX + 1] = {0, 1, 2};

void tiles()
{
	for (int i = 3; i <= n; i++)
		d[i] = (d[i - 1] + d[i - 2]) % MOD;
}

int main()
{
	scanf_s("%d", &n);

	tiles();

	printf("%d\n", d[n]);

	return 0;
}

File: DynamicProgramming1/01Tiles/Go01Tiles/01Tiles.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const (
	max = 1000000
	mod = 15746
)

var (
	n int
	d [max + 1]int
)

func tiles() {
	d[0] = 0
	d[1] = 1
	d[2] = 2

	for i := 3; i <= n; i++ {
		d[i] = (d[i-1] + d[i-2]) % mod
	}
}

func main() {
	br := bufio.NewReader(os.Stdin)
	bw := bufio.NewWriter(os.Stdout)

	fmt.Fscanln(br, &n)

	tiles()

	fmt.Fprintln(bw, d[n])
	bw.Flush()
}
